ibis/backends/dask/udf.py

```diff
@@ -0,0 +1,161 @@
+import itertools
+
+import dask.dataframe as dd
+import dask.dataframe.groupby as ddgb
+import dask.delayed
+import pandas
+
+import ibis.client
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.udf.vectorized
+from ibis.backends.pandas.udf import nullable  # noqa
+
+from .dispatch import execute_node, pre_execute
+from .execution.util import assert_identical_grouping_keys, make_selected_obj
+
+
+@pre_execute.register(ops.ElementWiseVectorizedUDF)
+@pre_execute.register(ops.ElementWiseVectorizedUDF, ibis.client.Client)
+def pre_execute_elementwise_udf(op, *clients, scope=None, **kwargs):
+    """Register execution rules for elementwise UDFs.
+    """
+    input_type = op.input_type
+
+    # definitions
+
+    # Define an execution rule for elementwise operations on a
+    # grouped Series
+    nargs = len(input_type)
+
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF,
+        *(itertools.repeat(ddgb.SeriesGroupBy, nargs)),
+    )
+    def execute_udf_node_groupby(op, *args, **kwargs):
+        func = op.func
+
+        # all grouping keys must be identical
+        assert_identical_grouping_keys(*args)
+
+        # we're performing a scalar operation on grouped column, so
+        # perform the operation directly on the underlying Series
+        # and regroup after it's finished
+        args_objs = [make_selected_obj(arg) for arg in args]
+        groupings = args[0].index
+        return dd.map_partitions(func, *args_objs).groupby(groupings)
+
+    # Define an execution rule for a simple elementwise Series
+    # function
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF, *(itertools.repeat(dd.Series, nargs))
+    )
+    def execute_udf_node(op, *args, **kwargs):
+        # We have rewritten op.func to be a closure enclosing
+        # the kwargs, and therefore, we do not need to pass
+        # kwargs here. This is true for all udf execution in this
+        # file.
+        # See ibis.udf.vectorized.UserDefinedFunction
+        if isinstance(op._output_type, dt.Struct):
+            # dask does not know how to glue together Tuple[*pd.Series].
```

Comment:
is this a bug on the dask side?

Reply:
I don't know if this is a bug vs. us just using this in an unexpected way. 

```python

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[56] > /home/g/dev/gerrymanoim/ibis/ibis/backends/dask/udf.py(68)execute_udf_node()
-> df.index = args[0].index
   5 frames hidden (try 'help hidden_frames')
(Pdb++) df
Dask Series Structure:
npartitions=2
0       object
3650       ...
7299       ...
dtype: object
Dask Name: wrapper, 6 tasks
(Pdb++) df.compute()
0         (1.0, 2.0)
1       (11.1, 12.1)
2       (21.2, 22.2)
3       (31.3, 32.3)
4       (41.4, 42.4)
            ...     
3645    (51.5, 52.5)
3646    (61.6, 62.6)
3647    (71.7, 72.7)
3648    (81.8, 82.8)
3649    (91.9, 92.9)
Length: 7300, dtype: object
(Pdb++) dd.map_partitions(op.func, *args, meta="object")
Dask Series Structure:
npartitions=2
0       object
3650       ...
7299       ...
dtype: object
Dask Name: add_one_struct, 6 tasks
(Pdb++) dd.map_partitions(op.func, *args, meta="object").compute()
0    ([1.0, 11.1, 21.2, 31.3, 41.4, 51.5, 61.6, 71....
1    ([21.2, 31.3, 41.4, 51.5, 61.6, 71.7, 81.8, 91...
```

By wrapping we get a result of the right "shape"